File: conseil/src/main/java/com/example/conseil/entities/Article.java
package com.example.conseil.entities;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;
import java.time.LocalDateTime;


@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
     private String titre;
    @Column(nullable = false)
     private String contenu;
    @Column(nullable = false)
     private Date datePublication;
    @Lob
    @Column(length = 1000000)
    private byte[] image;
    @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "specialist_id")
     private Specialist specialist;
}

File: FrontEndAngular/src/app/models/article.ts

export class Article {
  id: number;
  titre: string;
  contenu: string;
  datePublication: Date; // Use Date for LocalDateTime equivalent
  image?: string; // Optional, as it is nullable in your entity
  specialist_id: number; // Representing the relation with Specialist
  comment_id:number;

  constructor(
    id: number,
    titre: string,
    contenu: string,
    datePublication: Date,
    image?: string,
    specialist_id?: number,
    comment_id?:number
  ) {
    this.id = id;
    this.titre = titre;
    this.contenu = contenu;
    this.datePublication = datePublication;
    this.image = image || '';
    this.specialist_id = specialist_id || 0;
    this.comment_id= comment_id || 0;
  }
}
